packages/gitbook/src/components/primitives/Input.tsx

```diff
@@ -0,0 +1,313 @@
+                              'focus-within:border-primary-hover focus-within:depth-subtle:shadow-lg focus-within:shadow-primary-subtle focus-within:ring-2 hover:cursor-text hover:border-tint-hover depth-subtle:hover:not-focus-within:shadow-md focus-within:hover:border-primary-hover',
+                          ],
+                    multiline ? 'flex-col' : 'flex-row',
+                    ariaBusy ? 'cursor-progress' : '',
+                    sizes[sizing].container,
+                    className
+                )}
+                onClick={handleClick}
+                onKeyDown={(event) => {
+                    if (event.key === 'Enter' || event.key === ' ') {
+                        handleClick();
+                    }
+                }}
+                ref={containerRef}
+            >
+                <Tag
+                    className={tcls(
+                        'flex grow',
+                        sizes[sizing].gap,
+                        multiline ? 'items-start' : 'items-center'
+                    )}
+                >
+                    {leading ? (
+                        <Tag
+                            className={tcls(
+                                clearButton && hasValue ? 'group-focus-within/input:hidden' : '',
+                                multiline ? 'my-1.25' : ''
+                            )}
+                        >
+                            {typeof leading === 'string' ? (
+                                <Icon
+                                    icon={leading as IconName}
+                                    className="size-4 shrink-0 text-tint"
+                                />
+                            ) : (
+                                leading
+                            )}
+                        </Tag>
+                    ) : null}
+                    {clearButton ? (
+                        <Button
+                            variant="blank"
+                            size="medium"
+                            label={tString(language, 'clear')}
+                            iconOnly
+                            icon="circle-xmark"
+                            onClick={handleClear}
+                            {...(typeof clearButton === 'object' ? clearButton : {})}
+                            className={tcls(
+                                '-mx-1.5 hidden shrink-0 animate-fade-in p-1.5 text-tint',
+                                multiline ? '-my-0.25' : '-my-1.5',
+                                hasValue ? 'group-focus-within/input:flex' : '',
+                                typeof clearButton === 'object' ? clearButton.className : ''
+                            )}
+                        />
+                    ) : null}
+                    {multiline ? (
+                        <textarea
+                            {...(inputProps as React.TextareaHTMLAttributes<HTMLTextAreaElement>)}
```

Comment:
You should not have to do `as`, if you have to it means something is wrong in terms of typing.